src/Type/QueryResultToArrayDynamicReturnTypeExtension.php

```diff
@@ -43,6 +44,8 @@ public function getTypeFromMethodCall(
 		$resultType = $scope->getType($methodCall->var);
 		if ($resultType instanceof GenericObjectType) {
 			$modelType = $resultType->getTypes();
+		} elseif($resultType instanceof CompoundType && $resultType->getTypes()[0] instanceof GenericObjectType) {
```

Comment:
This is IMO not a really nice solution but works for now. If someone has time to investigate how to make it more stable feel free to contribute.

Reply:
I've adjusted this bit.